File: Examples/lua/hello.lua
-- =============================================
--  Lua Comprehensive Demo - Time Warp Studio
-- =============================================

-- --- Hello World ---
print("===== HELLO WORLD =====")
print("Welcome to Lua!")
print("")

-- --- Variables ---
print("===== VARIABLES =====")
local x = 10
local y = 3
local pi = 3.14159
local name = "Lua"
local flag = true
print("int: " .. x)
print("float: " .. pi)
print("string: " .. name)
print("bool: " .. tostring(flag))
print("")

-- --- Arithmetic ---
print("===== ARITHMETIC =====")
print("x + y = " .. (x + y))
print("x - y = " .. (x - y))
print("x * y = " .. (x * y))
print("x / y = " .. (x / y))
print("x % y = " .. (x % y))
print("2 ^ 10 = " .. (2 ^ 10))
print("")

-- --- Strings ---
print("===== STRINGS =====")
print("upper: " .. string.upper("hello"))
print("lower: " .. string.lower("HELLO"))
print("rep: " .. string.rep("ab", 3))
print("sub: " .. string.sub("Hello World", 1, 5))
print("len: " .. string.len("Hello"))
print("reverse: " .. string.reverse("Hello"))
print("concat: " .. "Hello" .. " " .. "World")
print("")

-- --- Tables (Arrays) ---
print("===== TABLES =====")
local fruits = {"apple", "banana", "cherry"}
print("first: " .. fruits[1])
print("second: " .. fruits[2])
print("third: " .. fruits[3])
print("count: " .. #fruits)
print("joined: " .. table.concat(fruits, ", "))
print("")

-- --- Tables (Dictionaries) ---
print("===== DICT TABLES =====")
local person = {name = "Alice", age = 30, city = "Wonderland"}
print("name: " .. person.name)
print("age: " .. person.age)
print("city: " .. person.city)
print("")

-- --- Control Flow ---
print("===== CONDITIONALS =====")
if x > 5 then
  print("x is greater than 5")
elseif x == 5 then
  print("x equals 5")
else
  print("x is less than 5")
end
print("")

-- --- For Loop (numeric) ---
print("===== FOR LOOP =====")
for i = 1, 5 do
  print(i)
end
print("")

-- --- For Loop (generic) ---
print("===== GENERIC FOR =====")
for i, v in ipairs(fruits) do
  print(i .. ": " .. v)
end
print("")

-- --- While Loop ---
print("===== WHILE LOOP =====")
local n = 1
while n <= 3 do
  print("n = " .. n)
  n = n + 1
end
print("")

-- --- Repeat-Until ---
print("===== REPEAT-UNTIL =====")
local m = 1
repeat
  print("m = " .. m)
  m = m + 1
until m > 3
print("")

-- --- Functions ---
print("===== FUNCTIONS =====")
function factorial(n)
  if n <= 1 then return 1 end
  return n * factorial(n - 1)
end
print("5! = " .. factorial(5))
print("10! = " .. factorial(10))

function greet(who)
  return "Hello " .. who
end
print(greet("World"))
print("")

-- --- Math ---
print("===== MATH =====")
print("sqrt(16) = " .. math.sqrt(16))
print("pi = " .. math.pi)
print("floor(3.7) = " .. math.floor(3.7))
print("ceil(3.2) = " .. math.ceil(3.2))
print("abs(-5) = " .. math.abs(-5))
print("max(3,7) = " .. math.max(3, 7))
print("min(3,7) = " .. math.min(3, 7))
print("")

-- --- Type Checking ---
print("===== TYPES =====")
print("type(42) = " .. type(42))
print("type('hi') = " .. type("hi"))
print("type(true) = " .. type(true))
print("type({}) = " .. type({}))
print("")

print("===== DONE =====")

File: Examples/c/hello.c
/* =============================================
 *  C Comprehensive Demo - Time Warp Studio
 * ============================================= */
#include <stdio.h>

int main() {
    /* --- Hello World --- */
    printf("===== HELLO WORLD =====\n");
    printf("Welcome to C!\n");
    printf("\n");

    /* --- Variables and Types --- */
    printf("===== VARIABLES & TYPES =====\n");
    int x = 10;
    int y = 3;
    float pi = 3.14159;
    char ch = 'A';
    printf("int: %d\n", x);
    printf("float: %f\n", pi);
    printf("char: %c\n", ch);
    printf("\n");

    /* --- Arithmetic --- */
    printf("===== ARITHMETIC =====\n");
    printf("x + y = %d\n", x + y);
    printf("x - y = %d\n", x - y);
    printf("x * y = %d\n", x * y);
    printf("x / y = %d\n", x / y);
    printf("x %% y = %d\n", x % y);
    x++;
    printf("x++ = %d\n", x);
    x--;
    printf("x-- = %d\n", x);
    printf("\n");

    /* --- Conditionals --- */
    printf("===== CONDITIONALS =====\n");
    if (x > 5) {
        printf("x is greater than 5\n");
    } else {
        printf("x is not greater than 5\n");
    }

    if (y == 3) {
        printf("y equals 3\n");
    }
    printf("\n");

    /* --- For Loop --- */
    printf("===== FOR LOOP =====\n");
    for (int i = 1; i <= 5; i++) {
        printf("%d\n", i);
    }
    printf("\n");

    /* --- While Loop --- */
    printf("===== WHILE LOOP =====\n");
    int count = 0;
    while (count < 3) {
        printf("count = %d\n", count);
        count++;
    }
    printf("\n");

    /* --- Do-While Loop --- */
    printf("===== DO-WHILE LOOP =====\n");
    int n = 1;
    do {
        printf("n = %d\n", n);
        n++;
    } while (n <= 3);
    printf("\n");

    /* --- Format Specifiers --- */
    printf("===== FORMAT SPECIFIERS =====\n");
    printf("Hex: %x\n", 255);
    printf("Oct: %o\n", 255);
    printf("Sci: %e\n", 12345.6789);
    printf("Percent: 100%%\n");
    printf("\n");

    /* --- String Functions --- */
    printf("===== STRINGS =====\n");
    printf("strlen: %d\n", strlen("Hello"));
    char dest[30];
    strcpy(dest, "Hello");
    strcat(dest, " World");
    printf("strcat: %s\n", dest);
    printf("\n");

    /* --- Math Functions --- */
    printf("===== MATH =====\n");
    printf("abs(-5) = %d\n", abs(-5));
    printf("sqrt(16) = %f\n", sqrt(16.0));
    printf("pow(2, 10) = %f\n", pow(2.0, 10.0));
    printf("\n");

    printf("===== DONE =====\n");
    return 0;
}
